Scripts/Brushes/PrimitiveBrush.cs

```diff
@@ -817,7 +817,7 @@ public void ResetPivot()
 			}
 
 			// Bounds is aligned with the object
-			transform.Translate(delta);
+			transform.Translate(delta.Multiply(transform.localScale));
```

Comment:
Conceivably this could be lossyScale, but that may have issues if the brush is rotated and the parent/root is scaled non-uniformly

Reply:
I had the same thoughts and lossyScale worked just fine on my quick tests however like I mentioned the parent/root scaling is ignored by the CSG building process (the tools and wire-frames scale properly), it's not ignored for brushes where it works perfectly but it's ignored again for compound brushes (another parent scale I suppose). LossyScale did not resolve this problem in any way or form. It would be nice to fix all of the scaling. But unless you immediately know where to look to change the CSG output to scale according to the parents...?